### PR TITLE
#112 add weekly triggered ci job

### LINKED ISSUE #112: Add weekly triggered CI Job
Add CI job which is runs once a week, to ensure the project still builds and runs with all the unpinned SW versions.

_Originally posted by @tkilias in https://github.com/exasol/sqlalchemy-exasol/issues/111#issuecomment-1057920757_

### PR DESCRIPTION
**Pull request by Nicoretti**
Fixes #112 

- Add weekly scheduled CI/CD run
- Pin ITDE version in github action(s)

### CODE REVIEW

**Review by pj-spoelders — DISMISSED** [review]

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -55,8 +55,11 @@ jobs:
       if: ${{ matrix.connector == 'turbodbc' }}
 
     - name: Checkout test environment
-      run: git clone https://github.com/exasol/integration-test-docker-environment.git
+      run: git clone --depth 1 --branch ${ITDE_TAG} ${ITDE_URL}
```
Comment (tkilias) [inline .github/workflows/CI.yml:61]: nice, in general, I currently suggest to add a shell script which starts the itde, you can use the same for local setup

Comment (tkilias) [reply .github/workflows/CI.yml:61]: like here https://github.com/exasol/advanced-analytics-framework/blob/main/scripts/start_integration_test_environment.sh

Comment (Nicoretti) [reply .github/workflows/CI.yml:61]: > nice, in general, I currently suggest to add a shell script which starts the itde, you can use the same for local setup

nox currently starts the itde not sure though if it also should take care of the version of the dependency. I would assume the default for the local dev machine is main anyway and for the CI it should be pinned.

Or is there critical issue with this in your opinion?

Comment (tkilias) [reply .github/workflows/CI.yml:61]: Ah you are right there was an automation. In general, I would suggest to pin the version, to decouple things

**Review by pj-spoelders — APPROVED** [review]

**Pull request merged**